alpha/apps/kaltura/lib/storage/kFileUtils.php

```diff
@@ -105,20 +105,29 @@ public static function dumpApiRequest($host, $onlyIfAvailable = false)
 		
 		$url = $_SERVER['REQUEST_URI'];
 		if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on' && kConf::hasParam('https_param_salt'))
+			$post_params['apiProtocol'] = 'https_' . kConf::get('https_param_salt');
+		
+		if(isset($_SERVER['CONTENT_TYPE']))
 		{
-			$concatStr = strpos($url, "?") !== false ? "&" : "?";
-			$url = $url . $concatStr . 'apiProtocol=https_' . kConf::get('https_param_salt');
+			if(strtolower($_SERVER['CONTENT_TYPE']) == 'application/json' || (strpos(strtolower($_SERVER['CONTENT_TYPE']), 'multipart/form-data') === 0 && isset($_POST['json'])))
+			{
+				$post_params = array_merge($post_params, infraRequestUtils::getRequestParams());
+			}
 		}
-			
-		$httpHeader = array("X-Kaltura-Proxy: dumpApiRequest");
 		
-		if(isset($_SERVER['CONTENT_TYPE']))
+		foreach($post_params as $key => $value)
 		{
-			$httpHeader[] = "Content-Type: " . $_SERVER['CONTENT_TYPE'];
-				
-			if(strtolower($_SERVER['CONTENT_TYPE']) == 'application/json')
-				$post_params = infraRequestUtils::$phpInputStream;
+			if(is_array($value))
+			{
+				foreach($value as $valueKey => $valueVal)
+				{
+					$post_params["$key:$valueKey"] = $valueVal;
```

Comment:
That should be done using recursion as in most cases the array will contain objects that may also contain nested objects.